pandas/io/parsers.py

```diff
@@ -1755,6 +1755,10 @@ def _cast_types(self, values, cast_type, column):
                 cats, cats.get_indexer(values), cast_type,
                 true_values=self.true_values)
 
+        elif is_bool_dtype(cast_type) and any(isna(values)):
+            raise ValueError("Bool column has NA values in "
+                             "column {column}".format(column=column))
+
```

Comment:
This looks okay...I'm wondering though if you can make it work in `self._convert_data`.  That would be preferable.  No worries if you can't, but if you could take a look, that would be great.